### PR TITLE
Simd quantize fp

### PR DESCRIPTION
**Pull request by lzhangnj**
Port av1_quantize_fp_facade() avx2 from aom

### CODE REVIEW

**Comment by lzhangnj** [comment]
To activate and test, please change "EbBool perform_quantize_fp = EB_FALSE;" to "EbBool perform_quantize_fp = EB_TRUE;" in av1_quantize_inv_quantize().

**Comment by hassount** [comment]
@ttrigui @lzhangnj  can you please enable this in enc_mo after checking bitstream exactness ?

**Comment by ttrigui** [comment]
@hassount can you please enable this in enc_mo after checking bitstream exactness ? => done

**Comment by lzhangnj** [comment]
Related to Issue #262

**Comment by mafonsoN** [comment]
Do you have any numbers on how much this increases the speed for each encoding mode?

**Comment by hassount** [comment]
@mafonsoN the feature is off by default as it is written in C, this PR makes the feature cost ~0% in -enc-mode 0 being the only preset where it's used for now.

File: Source/Lib/Common/ASM_AVX2/av1_quantize_avx2.c [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+    if (sizeof(TranLow) == 4) {
+        const __m256i x0 = _mm256_loadu_si256((const __m256i *)coeff);
+        const __m256i x1 = _mm256_loadu_si256((const __m256i *)coeff + 1);
+        *c = _mm256_packs_epi32(x0, x1);
+        *c = _mm256_permute4x64_epi64(*c, 0xD8);
+    }
+    else {
+        *c = _mm256_loadu_si256((const __m256i *)coeff);
+    }
+}
```
Comment (joelsole) [inline Source/Lib/Common/ASM_AVX2/av1_quantize_avx2.c]: no braces

**Comment by mafonsoN** [comment]
@hassount does enabling this feature give any BD-rate gains then compared to the RDOQ?

**Comment by hassount** [comment]
@mafonsoN 
yes 0.3% gain and is explained in #341 at the description: https://github.com/OpenVisualCloud/SVT-AV1/pull/341#issue-285687449 

Sorry we should have probably made the link earlier, it was only made through #262

**Comment by EwoutH** [comment]
@joelsole Are all your discussions resolved? If yes, could you mark as approved?

@ttrigui Any comments?

**Comment by hassount** [comment]
Hi @lzhangnj 
the style check on this branch is failing, such could be due to the base of the branch. would you be able to rebase the branch so that we could check whether it passes or not ?

**Comment by lzhangnj** [comment]
> Hi @lzhangnj
> the style check on this branch is failing, such could be due to the base of the branch. would you be able to rebase the branch so that we could check whether it passes or not ?

Done.

**Pull request merged**